src/jit/hwintrinsicxarch.cpp

```diff
@@ -466,6 +479,14 @@ GenTree* Compiler::impAVXIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HAND
             retNode  = gtNewSimdHWIntrinsicNode(TYP_SIMD32, op1, op2, NI_AVX_Add, baseType, 32);
             break;
 
+        case NI_AVX_Multiply:
+            assert(sig->numArgs == 2);
+            op2      = impSIMDPopStack(TYP_SIMD32);
+            op1      = impSIMDPopStack(TYP_SIMD32);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            retNode  = gtNewSimdHWIntrinsicNode(TYP_SIMD32, op1, op2, NI_AVX_Multiply, baseType, 32);
```

Comment:
You should pass `intrinsic` instead of `NI_AVX_Multiply` to `gtNewSimdHWIntrinsicNode`, then the code for `Add` and `Multiply` will be identical..